src/pb_cancel.cc

```diff
@@ -57,6 +57,9 @@ PbCancel::ShmPayload()
 bool
 PbCancel::IsCancelled()
 {
+  // Release the GIL. Python objects are not accessed during the check.
```

Comment:
If it's not too much work, can we elaborate performance benefits in the comment above or in the PR description?

Reply:
Sure added to the PR description.